python/cuml/preprocessing/TargetEncoder.py

```diff
@@ -0,0 +1,213 @@
+        res, train = self._fit_transform(x, y)
+        self.train_encode = res
+        self.train = train
+        self._fitted = True
+        return self
+
+    def fit_transform(self, x, y):
+        self.fit(x, y)
+        return self.train_encode
+
+    def transform(self, x):
+        self._check_is_fitted()
+        test = self._to_frame(x)
+        if self._is_train_df(test):
+            return self.train_encode
+        x_cols = [i for i in test.columns.tolist() if i != self.id_col]
+        test = test.merge(self.agg_all, on=x_cols, how='left')
+        return self._get_return_value(test)
+
+    def _fit_transform(self, x, y):
+        cp.random.seed(self.seed)
+        train = self._to_frame(x)
+        x_cols = [i for i in train.columns.tolist() if i != self.id_col]
+
+        if isinstance(y, gd.Series):
+            train[self.y_col] = y.values
+        elif isinstance(y, cp.ndarray):
+            if len(y.shape) == 1:
+                train[self.y_col] = y
+            elif y.shape[1] == 1:
+                train[self.y_col] = y[:, 0]
+            else:
+                raise ValueError(f"Input of shape {y.shape} "
+                                 "is not a 1-D array.")
+        else:
+            raise TypeError(
+                "Input of type {type(y)} is not cudf.Series, "
+                "or cupy.ndarray")
+
+        if self.split == 'random':
+            train[self.fold_col] = cp.random.randint(0,
+                                                     self.n_folds, len(train))
+        elif self.split == 'continuous':
+            train[self.fold_col] = cp.arange(len(train)) / \
+                (len(train)/self.n_folds)
+            train[self.fold_col] = train[self.fold_col] % self.n_folds
+        elif self.split == 'interleaved':
+            train[self.fold_col] = cp.arange(len(train))
+            train[self.fold_col] = train[self.fold_col] % self.n_folds
+        else:
+            msg = ("split should be either 'random'"
+                   " or 'continuous' or 'interleaved', "
+                   "got {0}.".format(self.split))
+            raise ValueError(msg)
+
+        train[self.fold_col] = train[self.fold_col].astype('int32')
+        self.mean = train[self.y_col].mean()
+
+        cols = [self.fold_col]+x_cols
+
+        agg_each_fold = train.groupby(cols).agg({self.y_col:
+                                                 ['count',
+                                                  'sum']}).reset_index()
+        agg_each_fold.columns = cols + ['count_y', 'sum_y']
+
+        agg_all = agg_each_fold.groupby(x_cols).agg({'count_y': 'sum',
+                                                     'sum_y': 'sum'})\
+                                               .reset_index()
+        cols = x_cols
+        agg_all.columns = cols + ['count_y_all', 'sum_y_all']
+
+        agg_each_fold = agg_each_fold.merge(agg_all, on=x_cols, how='left')
+        agg_each_fold['count_y_all'] = agg_each_fold['count_y_all'] - \
```

Comment:
do these all need to be columns in a df? Seems like each could just be its own series-style groupby output, like:
```
 count_y_all = ... aggregate here ...
```

I think that would make the code clearer and eliminate the `drop` step below

Reply:
Should be done. Thanks.